server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -99,12 +105,24 @@ public class CoordinationDiagnosticsService implements ClusterStateListener {
     // Non-private for testing
     volatile ConcurrentMap<DiscoveryNode, ClusterFormationStateOrException> clusterFormationResponses = null;
 
+    /*
+     * This is a reference to the task that is periodically reaching out to a master eligible node to get its CoordinationDiagnosticsResult
+     * for diagnosis. It is null when no polling is occurring.
+     * The field is accessed (reads/writes) from multiple threads, and is also reassigned on multiple threads.
+     */
+    volatile AtomicReference<Scheduler.Cancellable> remoteCoordinationDiagnosisTask = null;
+    /*
+     * This field holds the result of the task in the remoteCoordinationDiagnosisTask field above. The field is accessed
+     * (reads/writes) from multiple threads, but is only ever reassigned on a single thread (the cluster change event thread).
+     */
+    volatile AtomicReference<RemoteMasterHealthResult> remoteCoordinationDiagnosisResult = null;
+
     /**
      * This is the amount of time that we wait before scheduling a remote request to gather diagnostic information. It is not
      * user-configurable, but is non-final so that integration tests don't have to waste 10 seconds.
      */
     // Non-private for testing
-    TimeValue remoteRequestInitialDelay = new TimeValue(10, TimeUnit.SECONDS);
+    public TimeValue remoteRequestInitialDelay = new TimeValue(10, TimeUnit.SECONDS);
```

Comment:
I think we can keep this `default` visible if we move `StableMasterDisruptionIT` to the `org.elasticsearch.cluster.coordination` package (it doesn't really need to be under `discovery` - and the `cluster.coordination` package is better suited given the indicator is under the same package too)

What do you think?

Reply:
I'll take this out of this PR and do that in another one since it's unrelated to this work.

Reply:
Sounds good, thanks Keith